flink-runtime/src/main/java/org/apache/flink/runtime/deployment/InputGateDeploymentDescriptor.java

```diff
@@ -59,17 +62,34 @@
     @Nonnegative private final int consumedSubpartitionIndex;
 
     /** An input channel for each consumed subpartition. */
-    private final ShuffleDescriptor[] inputChannels;
+    private transient ShuffleDescriptor[] inputChannels;
+
+    /** Serialized value of shuffle descriptors. */
+    private final CompressedSerializedValue<ShuffleDescriptor[]> serializedInputChannels;
```

Comment:
I think declare it as `SerializedValue<ShuffleDescriptor[]>` is enough.
Whether to compress it is an implementation detail.

Reply:
Thank you for pointing this out. I've changed it into `SerializedValue`.